examples/frigate/frigate.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+import dataclasses
+from datetime import datetime
+import typing as t
+
+from mqttwarn.context import RuntimeContext
+from mqttwarn.model import Service
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+
+@dataclasses.dataclass
+class FrigateEvent:
+    """
+    Manage inbound event data received from Frigate.
+    """
+    time: datetime
+    camera: str
+    label: str
+    current_zones: t.List[str]
+    entered_zones: t.List[str]
+
+    def f(self, value):
+        return [y.replace('_', ' ') for y in value]
+
+    @property
+    def current_zones_str(self):
+        return ', '.join(self.f(self.current_zones))
+
+    @property
+    def entered_zones_str(self):
+        return ', '.join(self.f(self.entered_zones))
+
+    def to_dict(self) -> t.Dict[str, str]:
+        return dataclasses.asdict(self)
+
+
+@dataclasses.dataclass
+class NtfyParameters:
+    """
+    Manage outbound parameter data for Apprise/Ntfy.
+    """
+    title: str
+    format: str
+    click: str
+    attach: str
+
+    def to_dict(self) -> t.Dict[str, str]:
+        return dataclasses.asdict(self)
+
+
+def frigate_events(topic, data, srv: Service):
+    """
+    mqttwarn transformation function which computes options to be submitted to Apprise/Ntfy.
+    """
+
+    # Acceptable hack to get attachment filename template from service configuration.
+    context: RuntimeContext = srv.mwcore["context"]
+    service_config = context.get_service_config("apprise-ntfy")
+    filename_template = service_config.get("filename_template")
+
+    # Decode JSON message.
+    after = json.loads(data['payload'])['after']
+
+    # Collect details from inbound Frigate event.
+    event = FrigateEvent(
+        time=datetime.fromtimestamp(after['frame_time']),
+        camera=after['camera'],
+        label=after['sub_label'] or after['label'],
+        current_zones=after['current_zones'],
+        entered_zones=after['entered_zones'],
+    )
+
+    # Interpolate event data into attachment filename template.
+    attach_filename = filename_template.format(**event.to_dict())
+
+    # Compute parameters for outbound Apprise / Ntfy URL.
+    ntfy_parameters = NtfyParameters(
+        title=f"{event.label} entered {event.entered_zones_str}",
+        format=f"In zones {event.current_zones_str} at {event.time}",
```

Comment:
I revised this in my own configuration to make it read better when `{event.current_zones_str}` is null. But the current one may work better for example as it is simpler.
```suggestion
        format=f"At {event.time}{', was in ' + event.current_zones_str if event.current_zones else ''}",
```